From b67a6184ed3b6d728894eba37a554a302c1b0312 Mon Sep 17 00:00:00 2001
From: "yong.gan" <yong.gan@nxp.com>
Date: Sat, 2 Apr 2016 09:33:56 +0800
Subject: [PATCH 3/3] MGS-1724: xwld: G2D compositor build failed in slevk
 board

Add macro ENABLE_EGL to make sure the EGL was not built in slevk board.
Modify the wrong format for the shm buffer.

Upstream Status: Inappropriate [i.MX specific]

Signed-off-by: Yong Gan <yong.gan@nxp.com>
---
 src/compositor-fbdev.c | 11 ++++++++++-
 src/g2d-renderer.c     |  4 ++--
 2 files changed, 12 insertions(+), 3 deletions(-)

Index: weston-1.10.0/src/compositor-fbdev.c
===================================================================
--- weston-1.10.0.orig/src/compositor-fbdev.c	2016-05-17 14:17:37.039727929 -0500
+++ weston-1.10.0/src/compositor-fbdev.c	2016-05-17 14:20:36.000000000 -0500
@@ -61,7 +61,9 @@
 	int use_pixman;
 	int use_g2d;
 	struct wl_listener session_listener;
+#ifdef ENABLE_EGL
 	NativeDisplayType display;
+#endif
 };
 
 struct fbdev_screeninfo {
@@ -94,9 +96,10 @@
 	/* pixman details. */
 	pixman_image_t *hw_surface;
 	uint8_t depth;
-
+#ifdef ENABLE_EGL
 	NativeDisplayType display;
 	NativeWindowType  window;
+#endif
 };
 
 struct fbdev_parameters {
@@ -455,10 +458,12 @@
 		           strerror(errno));
 
 	output->fb = NULL;
+#ifdef ENABLE_EGL
 	if(output->window)
 		fbDestroyWindow(output->window);
 	if(output->display)
 		fbDestroyDisplay(output->display);
+#endif
 }
 
 static void fbdev_output_destroy(struct weston_output *base);
@@ -542,7 +547,7 @@
 			weston_log("g2d_renderer_output_create failed.\n");
 			goto out_hw_surface;
 		}
-
+#ifdef ENABLE_EGL
 	} else {
 		setenv("HYBRIS_EGLPLATFORM", "wayland", 1);
 		output->window = fbCreateWindow(backend->display, -1, -1, 0, 0);
@@ -557,6 +562,7 @@
 			weston_log("gl_renderer_output_create failed.\n");
 			goto out_hw_surface;
 		}
+#endif
 	}
 
 	loop = wl_display_get_event_loop(backend->compositor->wl_display);
@@ -860,6 +866,7 @@
 		}
 	}
 	 else {
+#ifdef ENABLE_EGL
 		gl_renderer = weston_load_module("gl-renderer.so",
 						 "gl_renderer_interface");
 		if (!gl_renderer) {
@@ -879,6 +886,7 @@
 			weston_log("gl_renderer_create failed.\n");
 			goto out_launcher;
 		}
+#endif
 	}
 	if(!backend->use_g2d)
 		if (fbdev_output_create(backend, 0, 0, param->device) < 0)
Index: weston-1.10.0/src/g2d-renderer.c
===================================================================
--- weston-1.10.0.orig/src/g2d-renderer.c	2016-05-17 14:17:37.039727929 -0500
+++ weston-1.10.0/src/g2d-renderer.c	2016-05-17 14:17:37.035727909 -0500
@@ -756,11 +756,11 @@
 
 	switch (wl_shm_buffer_get_format(shm_buffer)) {
 	case WL_SHM_FORMAT_XRGB8888:
-		g2dFormat = G2D_XRGB8888;
+		g2dFormat = G2D_BGRX8888;
 		gs->bpp = 4;
 		break;
 	case WL_SHM_FORMAT_ARGB8888:
-		g2dFormat = G2D_ARGB8888;
+		g2dFormat = G2D_BGRA8888;
 		gs->bpp = 4;
 		break;
 	case WL_SHM_FORMAT_RGB565:
